src/utils/index.ts

```diff
@@ -7,3 +7,27 @@ import { type Generic } from 'types/components';
  */
 export const isJSX = <T>(arg: T | Generic): arg is Generic =>
   React.isValidElement(arg);
+
+export const wrapHandleExitAction =
+  (
+    href: string,
+    onClick?: () => void,
+    onExit?: (exitAction: () => void) => void
+  ) =>
+  (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    if (onExit) {
+      onExit(
+        onClick ??
+          (() => {
+            window.location.assign(href);
+          })
+      );
+    } else if (onClick) {
+      onClick();
+    } else {
+      window.location.assign(href);
+    }
+  };
```

Comment:
You can change line 19 to 30 to something like:
```js
    const exitAction =
      onClick ??
      (() => {
        window.location.assign(href);
      });
    if (onExit) onExit(exitAction);
    else exitAction();
```

Reply:
done